File: Ex3Num8.go
package main

import "fmt"

func main() {
	var n, sum int
	fmt.Print("Введите количество чисел: ")
	fmt.Scan(&n)

	for i := 0; i < n; i++ {
		var number int
		fmt.Printf("Введите число %d: ", i+1)
		fmt.Scan(&number)

		// Обрабатываем отрицательные числа
		if number < 0
		{
			number = -number
		}

		// Разбираем число по цифрам
		for number > 0 {
			digit := number % 10
			if digit%3 == 0 && digit != 0 {
				sum += digit
			}
			number /= 10
		}
	}

	fmt.Printf("Сумма цифр, делящихся на 3: %d\n", sum)
}

File: Ex3Num8.php
<?php

$sum = 0;

echo "Введите количество чисел: ";
$n = (int)fgets(STDIN);

for ($i = 1; $i <= $n; $i++) {
    echo "Введите число $i: ";
    $number = (int)fgets(STDIN);

    // Обрабатываем отрицательные числа
    $number = abs($number);

    // Разбираем число по цифрам
    $temp = $number;
    while ($temp > 0) {
        $digit = $temp % 10;
        if ($digit % 3 == 0 && $digit != 0) {
            $sum += $digit;
        }
        $temp = (int)($temp / 10);
    }
}

echo "Сумма цифр, делящихся на 3: $sum\n";
?>